internal/pkg/service/objectstorage/cloud/aws/service.go

```diff
@@ -27,28 +28,33 @@ func (s AWSObjectStorageService) NewAccessRoleService(ctx context.Context, logge
 	}
 
 	// Assume role
+	awsCredentials, ok := cluster.GetCredentials().(AWSCredentials)
+	if !ok {
+		return nil, errors.New("Impossible to cast cluster credentials into AWS cluster credentials")
+	}
 	stsClient := sts.NewFromConfig(cfg)
-	credentials := stscreds.NewAssumeRoleProvider(stsClient, cluster.GetRole())
+	credentials := stscreds.NewAssumeRoleProvider(stsClient, awsCredentials.Role)
 	cfg.Credentials = aws.NewCredentialsCache(credentials)
 
-	parsedRole, err := arn.Parse(cluster.GetRole())
+	parsedRole, err := arn.Parse(awsCredentials.Role)
 	if err != nil {
 		return nil, errors.WithStack(err)
 	}
 
-	return NewIamService(iam.NewFromConfig(cfg), logger, parsedRole.AccountID, cluster), nil
+	return NewIamService(iam.NewFromConfig(cfg), logger, parsedRole.AccountID, cluster.(AWSCluster)), nil
```

Comment:
Maybe you should ensure this is an awscluster using awscluster, ok := cluster.(AWSCluster) ?